Sources/Basic/OutputByteStream.swift

```diff
@@ -55,32 +58,108 @@ public class OutputByteStream: OutputStream {
         return buffer.count
     }
 
-    /// The contents of the output stream.
-    ///
-    /// This method implicitly flushes the stream.
-    public var bytes: ByteString {
-        flush()
-        return ByteString(self.buffer)
+    /// Currently available buffer size.
+    private var availableBufferSize: Int {
+        return max(buffer.capacity, OutputByteStream.bufferSize) - buffer.count
     }
-    
+
+    /// Clears the buffer maintaining current capacity.
+    private func clearBuffer() {
+        buffer.removeAll(keepingCapacity: true)
+    }
+
     // MARK: Data Output API
 
     public func flush() {
-        // Do nothing.
+        writeImpl(ArraySlice(buffer))
```

Comment:
... because the base class should encode the logic of `flush == write then flush` and should be final, and the only thing subclasses can override is the "what happens on flush"